### PR TITLE
Fail fast when program creation fails

### PR DESCRIPTION
**Pull request by vdiez**

### CODE REVIEW

File: sonar-plugin/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/bridge/JsTsSensorTest.java [hunk 1]
```diff
@@ -526,25 +525,42 @@ void should_analyze_by_program() throws Exception {
...
+    );
+    var cause = assertThatThrownBy(() -> createSensor().execute(context))
+      .isInstanceOf(IllegalStateException.class)
+      .hasMessage("Analysis of JS/TS files failed")
+      .getCause();
+    verify(bridgeServerMock, times(1)).createProgram(captorProgram.capture());
+
+    cause.hasMessage(
+      String.format(
+        "Failed to create TypeScript program with TSConfig file %s. Highest TypeScript supported version is %s.",
```
Comment (ilia-kebets-sonarsource) [inline sonar-plugin/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/bridge/JsTsSensorTest.java:557]:
```suggestion
        "Failed to create a TypeScript program with the TSConfig file %s. The highest supported TypeScript version is %s.",
```

File: sonar-plugin/sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/bridge/AnalysisWithProgram.java [hunk 2]
```diff
@@ -79,15 +79,14 @@ void analyzeFiles(List<InputFile> inputFiles, List<String> tsConfigs) throws IOE
...
-                "Failed to create TypeScript program with TSConfig file %s. Highest TypeScript supported version is %s.",
-                tsConfig,
-                JavaScriptPlugin.TYPESCRIPT_VERSION
-              )
-            );
           PROFILER.stopInfo();
-          continue;
+          throw new IllegalStateException(
+            String.format(
+              "Failed to create TypeScript program with TSConfig file %s. Highest TypeScript supported version is %s.",
```
Comment (ilia-kebets-sonarsource) [inline sonar-plugin/sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/bridge/AnalysisWithProgram.java:85]:
```suggestion
              "Failed to create a TypeScript program with the TSConfig file %s. The highest supported TypeScript version is %s.",
```

**Review by ilia-kebets-sonarsource — APPROVED** [review]
2 small suggestions. I'm looking forward to delete this verbose mocked mess and move it to the bridge.

**Comment by vdiez** [comment]
Closing this until we have a better view of the possible impact and how many ways program creation can fail. -> CloudWatch logs stats

**Pull request closed without merging**